src/components/EmailForm.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import Form from "./reusables/Form";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import { FetchErrorData } from "opds-web-client/lib/interfaces";
+import { Store } from "redux";
+import { State } from "../reducers/index";
+import { LibraryData } from "../interfaces";
+import { CheckSoloIcon } from "@nypl/dgx-svg-icons";
+
+export interface EmailFormState {
+  sent: boolean;
+}
+
+export interface EmailFormOwnProps {
+  library: LibraryData;
+  store: Store<State>;
+}
+
+export interface EmailFormDispatchProps {
+  email: (data: FormData) => Promise<void>;
+}
+
+export interface EmailFormStateProps {
+  error?: FetchErrorData;
+}
+
+export interface EmailFormProps extends EmailFormOwnProps, EmailFormDispatchProps, EmailFormStateProps {}
+
+export class EmailForm extends React.Component<EmailFormProps, EmailFormState> {
+
+  constructor(props) {
+    super(props);
+    this.sendEmail = this.sendEmail.bind(this);
+    this.state = { sent: false };
+  }
+
+  async sendEmail(data: FormData): Promise<void> {
+    await this.props.email(data);
```

Comment:
This action only calls the server to send an email, but that action itself could return an error. You're catching that error in a state prop, but it seems like the state for `sent` will still be true. I don't see any place where it is being set to false so is it possible that `sent` will be true even though you get an error message from the server? I can't readily test this scenario and hope you have.

Reply:
My understanding is that, if the action results in an error, the subsequent code (i.e. the `setState` part) won't be run.  `LogInForm` does something very similar (line 43 et seq.)--if the `logIn` action fails, it will just show the error message, not go on to reload the page.  Wrong?

But at any rate, even if that's not the right explanation: I did test it, and it works the way it's supposed to; it does not set `sent` to true if there's an error message.

Reply:
Hm I figured it'd both go to the next step if it's successful or fails. But if it works...